src/v/coproc/api.cc

```diff
@@ -58,10 +59,11 @@ ss::future<> api::start() {
     co_await _mt_frontend.start_single(std::ref(_rs.topics_frontend));
     co_await _pacemaker.start(_engine_addr, std::ref(_rs));
     co_await _pacemaker.invoke_on_all(&coproc::pacemaker::start);
-    _listener = std::make_unique<wasm::event_listener>();
 
+    _listener = std::make_unique<wasm::event_listener>(_as);
```

Comment:
I think we should not pass as to event_listener from coproc::api.
event_listener is not only coproc listener. In future it will listen data_policy event, inline_wasm events...

Reply:
I think we need to modify how the event listener is instantiated. Something like
```
if(enable_coproc() || enable_v8()) {
  _listener = std::make_unique<wasm::event_listener(_as);
}
```